### PR TITLE
Missing publish / unpublish tests in InstructorHomePageUiTest #3671

### LINKED ISSUE #3671: Missing publish / unpublish tests in InstructorHomePageUiTest
It has been marked as a TODO for awhile, should implement such missing tests so it's easier to spot mistakes when we make functionality changes

### PR DESCRIPTION
**Pull request by kennho**
Fixes #3671

### CODE REVIEW

File: src/test/java/teammates/test/cases/ui/browsertests/InstructorHomePageUiTest.java [hunk 1]
```diff
@@ -296,8 +296,62 @@ public void testRemindActions(){
...
-        //TODO add test for publishing and unpublishing feedback sessions
+    public void testPublishUnpublishActions() {
+        ______TS("publish action: AWAITING feedback session");
+        
+        homePage.verifyUnclickable(homePage.getPublishLink(feedbackSession_AWAITING.courseId, feedbackSession_AWAITING.feedbackSessionName));
+
+        ______TS("publish action: OPEN feedback session");
+        
+        homePage.clickFeedbackSessionPublishLink(feedbackSession_OPEN.courseId, feedbackSession_OPEN.feedbackSessionName);
+        ThreadHelper.waitFor(1000);
```
Comment (kanghj) [inline src/test/java/teammates/test/cases/ui/browsertests/InstructorHomePageUiTest.java]: This is probably a bad idea... 

Try to find something on the page to wait for.

Comment (kennho) [inline src/test/java/teammates/test/cases/ui/browsertests/InstructorHomePageUiTest.java]: oh sorry was referring to other parts of the code when doing this. will change it! @kanghj

**Comment by kennho** [comment]
Updated suggested changes @kanghj

File: src/test/java/teammates/test/cases/ui/browsertests/InstructorHomePageUiTest.java [hunk 2]
```diff
@@ -296,8 +296,45 @@ public void testRemindActions(){
...
+        
+        homePage.clickFeedbackSessionUnpublishLink(feedbackSession_CLOSED.courseId, feedbackSession_CLOSED.feedbackSessionName);
+        homePage.waitForPageToLoad();
+        homePage.verifyStatus(Const.StatusMessages.FEEDBACK_SESSION_UNPUBLISHED);
+        
+        ______TS("unpublish action: PUBLISHED feedback session");
+        homePage.clickFeedbackSessionUnpublishLink(feedbackSession_PUBLISHED.courseId, feedbackSession_PUBLISHED.feedbackSessionName);
+        homePage.waitForPageToLoad();
+        homePage.verifyStatus(Const.StatusMessages.FEEDBACK_SESSION_UNPUBLISHED);
+        homePage.goToPreviousPage(InstructorHomePage.class);
```
Comment (kanghj) [inline src/test/java/teammates/test/cases/ui/browsertests/InstructorHomePageUiTest.java]: Is this needed? I thought the instructor remains on the same page after clicking publish or unpublish now.

Comment (kennho) [inline src/test/java/teammates/test/cases/ui/browsertests/InstructorHomePageUiTest.java]: yeah sorry missed that 1 out when removing it

**Comment by kennho** [comment]
Updated @kanghj

**Comment by kanghj** [comment]
The code looks fine to me, but doing it 3 times will cause quite a bit of emails to be sent out when testing the live server, probably contributing to our emails getting classified as spam.

**Comment by kennho** [comment]
Should I reduce it to do publish and unpublish 1 time each? @kanghj

**Comment by kanghj** [comment]
@kennho, sorry, but can you help me check how are publish and unpublish tested on the other pages? But yeah, I think doing it once should be enough, we can simplify the checks just to see if the 'publish'  and 'unpublish' button are clickable or not.

**Comment by kennho** [comment]
updated with suggested changes @kanghj

File: src/test/java/teammates/test/cases/ui/browsertests/InstructorHomePageUiTest.java [hunk 3]
```diff
@@ -296,8 +296,32 @@ public void testRemindActions(){
...
+
+        ______TS("publish action: CLOSED feedback session");
+        
+        homePage.clickAndCancel(homePage.getPublishLink(feedbackSession_CLOSED.courseId, feedbackSession_CLOSED.feedbackSessionName));
+        
+        ______TS("unpublish action: PUBLISHED feedback session");
+        homePage.clickFeedbackSessionUnpublishLink(feedbackSession_PUBLISHED.courseId, feedbackSession_PUBLISHED.feedbackSessionName);
+        homePage.waitForPageToLoad();
+        homePage.verifyStatus(Const.StatusMessages.FEEDBACK_SESSION_UNPUBLISHED);
+        assertEquals(false, BackDoor.getFeedbackSession(feedbackSession_PUBLISHED.courseId,
```
Comment (kanghj) [inline src/test/java/teammates/test/cases/ui/browsertests/InstructorHomePageUiTest.java]: Minor change, but use <code>assertFalse</code> here

File: src/test/java/teammates/test/cases/ui/browsertests/InstructorHomePageUiTest.java [hunk 4]
```diff
@@ -296,8 +296,32 @@ public void testRemindActions(){
...
+        homePage.waitForPageToLoad();
+        homePage.verifyStatus(Const.StatusMessages.FEEDBACK_SESSION_UNPUBLISHED);
+        assertEquals(false, BackDoor.getFeedbackSession(feedbackSession_PUBLISHED.courseId, 
+                                                        feedbackSession_PUBLISHED.feedbackSessionName).isPublished());
+
+        ______TS("publish action: PUBLISHED feedback session");
+        homePage.clickFeedbackSessionPublishLink(feedbackSession_PUBLISHED.courseId, feedbackSession_PUBLISHED.feedbackSessionName);
+        homePage.waitForPageToLoad();
+        homePage.verifyStatus(Const.StatusMessages.FEEDBACK_SESSION_PUBLISHED);
+        assertEquals(true, BackDoor.getFeedbackSession(feedbackSession_PUBLISHED.courseId,
```
Comment (kanghj) [inline src/test/java/teammates/test/cases/ui/browsertests/InstructorHomePageUiTest.java]: and use <code>assertTrue</code> here

**Comment by kennho** [comment]
Sorry missed the last comment. Updated with suggested changes @kanghj

**Pull request merged**